tests/e2e/common.go

```diff
@@ -130,40 +139,106 @@ func getNumaflowResourceStatus(u *unstructured.Unstructured) (kubernetes.Generic
 	return status, err
 }
 
-func getPodLogs(client clientgo.Interface, namespace, labelSelector, containerName, fileName string) {
-
-	ctx := context.Background()
-	podLogOptions := &corev1.PodLogOptions{Container: containerName}
-
-	podList, err := client.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+func watchPodLogs(client clientgo.Interface, namespace, labelSelector string) {
+	watcher, err := client.CoreV1().Pods(namespace).Watch(ctx, metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: "status.phase=Running"})
 	if err != nil {
 		fmt.Printf("Error listing pods: %v\n", err)
 		return
 	}
 
-	for _, pod := range podList.Items {
-		stream, err := client.CoreV1().Pods(namespace).GetLogs(pod.Name, podLogOptions).Stream(ctx)
-		if err != nil {
-			fmt.Printf("Error getting pods logs: %v\n", err)
+	for {
+		select {
+		case event := <-watcher.ResultChan():
+			if event.Type == watch.Modified || event.Type == watch.Added {
+				pod := event.Object.(*corev1.Pod)
+				for _, container := range pod.Spec.Containers {
+					streamPodLogs(context.Background(), kubeClient, Namespace, pod.Name, container.Name, stopCh)
```

Comment:
I see this can be called on `event.Type == watch.Modified` - does that imply it gets called for the same Pod more than once?

Reply:
Hm, I'm considering this might be causing an issue with our open files. Previously for the other watches I am using modified but that would make more sense in those cases as we want to keep updating on the state of the resource YAML. For this, we just want to be informed of new pods. Going to try a CI run with just Added.